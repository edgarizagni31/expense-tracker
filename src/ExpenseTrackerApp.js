import React, { useEffect, useState } from "react";
import { AddTransaction } from "./components/AddTransaction";
import { Balance } from "./components/Balance";
import { History } from "./components/History";

import './index.css';

export const ExpenseTrackerApp = () => {
  const initial =  () => JSON.parse(localStorage.getItem('transaction')) || [];
  const [transactions, setTransactions] = useState( initial );

  useEffect( () => {
    localStorage.setItem('transaction', JSON.stringify(transactions) );
  }, [transactions])
  
  return (
    <main className = 'mt-3 mx-auto main'>
      <h1 className = 'fs-4 fw-bold mb-3 text-center'> Registro de Gastos</h1>
      <Balance  transactions = { transactions }/>
      <History transactions = { transactions } setTransactions  = { setTransactions } />
      <AddTransaction setTransactions = { setTransactions } />
    </main>
  );
};
