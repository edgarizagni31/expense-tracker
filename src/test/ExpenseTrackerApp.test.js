import { shallow } from 'enzyme';
import { ExpenseTrackerApp } from '../ExpenseTrackerApp';

describe('test to <ExpenseTrackerApp />', () => {
    const wrapper = shallow(<ExpenseTrackerApp />);

    test('should show correctly', () => {
        expect(wrapper.find('Balance').exists() ).toBeTruthy();
        expect(wrapper.find('History').exists() ).toBeTruthy();
        expect(wrapper.find('AddTransaction').exists() ).toBeTruthy();
        expect(wrapper).toMatchSnapshot();        
    })
    
})