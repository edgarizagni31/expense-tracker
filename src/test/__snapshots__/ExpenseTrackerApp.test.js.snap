// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to <ExpenseTrackerApp /> should show correctly 1`] = `
<main
  className="mt-3 mx-auto main"
>
  <h1
    className="fs-4 fw-bold mb-3 text-center"
  >
     Registro de Transacciones 
  </h1>
  <Balance
    transactions={Array []}
  />
  <History
    setTransactions={[Function]}
    transactions={Array []}
  />
  <AddTransaction
    setTransactions={[Function]}
  />
</main>
`;
